pkg/controller/controller.go

```diff
@@ -108,11 +108,11 @@ func (c *controller) Run(workers int, stopCh <-chan struct{}) error {
 	var wg sync.WaitGroup
 	for i := 0; i < workers; i++ {
 		wg.Add(1)
-		// TODO (@munnerz): make time.Second duration configurable
-		go wait.Until(func() {
+		go func() {
 			defer wg.Done()
-			c.worker(ctx)
-		}, time.Second, stopCh)
+			// TODO (@munnerz): make time.Second duration configurable
```

Comment:
Can we remove this `TODO`? I can't think of a use case that would justify to make this configurable.

Reply:
Agreed on this.. :)

Reply:
After reading this block a second time, I do not understand why we should silently re-launch a mysteriously stopped worker:

```go
var wg sync.WaitGroup
for i := 0; i < workers; i++ {
	wg.Add(1)
	go func() {
		defer wg.Done()
		wait.Until(func() { c.worker(ctx) }, time.Second, stopCh) // 🍍 Silently re-launching the worker.
	}()
}
```

What is the rationale for silently re-starting? When/why the worker would stop? I would rather see cert-manager entirely stop after a worker returns.

Reply:
@maelvls I agree, I'll just remove the retry.

Reply:
I'd be cautious to remove this without completely understanding why it's there - from what I understand, it is to handle e.g. a panic causing this to exit. That said, I'm not 100% certain and I think this added in probably some of the first commits to the project, which were taken by copying/reading through the k/k codebase at the time

Reply:
As an example: https://github.com/kubernetes/kubernetes/blob/d9d4f0c69cbd3922dce7db924051f96e50524c38/pkg/controller/deployment/deployment_controller.go#L160-L162

Reply:
Seems like `wait.Until` will not prevent a crash; https://github.com/kubernetes/apimachinery/blob/f916759cb6b8547418dc7708876ecab5c1961448/pkg/util/wait/wait.go#L154

Reply:
FYI: @munnerz I think the line you refer to was added in https://github.com/kubernetes/kubernetes/pull/17398

Reply:
Also discussed here: https://github.com/jetstack/cert-manager/pull/3919#discussion_r622113559

Reply:
I think we agree that it should be deleted, as there is no panic recovery provided by `wait.Until`